master/internal/db/postgres_trial.go

```diff
@@ -163,23 +163,26 @@ WHERE id = $1`, id, restartCount); err != nil {
 func (db *PgDB) fullTrialSummaryMetricsRecompute(
 	ctx context.Context, tx *sqlx.Tx, trialID int,
 ) error {
-	trainSummary, err := db.calculateFullTrialSummaryMetrics(
-		ctx, tx, trialID, "avg_metrics", TrainingMetric)
-	if err != nil {
-		return fmt.Errorf("rollback computing training summary metrics: %w", err)
-	}
-	valSummary, err := db.calculateFullTrialSummaryMetrics(
-		ctx, tx, trialID, "validation_metrics", ValidationMetric)
-	if err != nil {
-		return fmt.Errorf("rollback computing validation summary metrics: %w", err)
-	}
-
 	updatedSummaryMetrics := model.JSONObj{}
-	if len(trainSummary) > 0 {
-		updatedSummaryMetrics["avg_metrics"] = trainSummary
-	}
-	if len(valSummary) > 0 {
-		updatedSummaryMetrics["validation_metrics"] = valSummary
+	metricTypes := []model.MetricType{}
+	if err := tx.SelectContext(ctx, &metricTypes,
+		`SELECT DISTINCT custom_type FROM metrics WHERE trial_id = $1`, trialID); err != nil {
```

Comment:
just scan the generic metrics table.

Reply:
add training and validation as a db constraints for generic metrics? and maybe check for reserved types there as well?